src/dsp_tools/commands/xmlupload/xmlupload.py

```diff
@@ -342,9 +347,17 @@ def _upload_one_resource(
     resource: XMLResource,
     ingest_client: AssetClient,
     resource_create_client: ResourceCreateClient,
-    lookups: Lookups,
+    iri_lookup: IRILookup,
+    intermediary_lookups: IntermediaryLookup,
     creation_attempts_of_this_round: int,
 ) -> None:
+    transformation_result = transform_into_intermediary_resource(resource, intermediary_lookups)
```

Comment:
I'm a bit surprised that you do this so late, immediately before the resource is uploaded. Wouldn't there be a much bigger benefit if this is done earlier, before the upload starts?

Reply:
Absolutely, I think I mentioned it somewhere, the reason why I do not, is because the diff in this PR is already quite large. Moving it out will generate an even larger diff. I wanted to break it up into two to make the review easier.